This patch uses fdwalk(3c) to close file descriptors; as that function is not
widely implemented, this is unsuitable for upstream.

Additional fix:
http://bugs.python.org/issue17964 was fixed in 3.4 but not back-ported so we
need this patch for 2.7 .

--- Python-2.7.15/Modules/posixmodule.c.orig	2018-11-13 04:01:30.469697707 -0800
+++ Python-2.7.15/Modules/posixmodule.c	2018-11-13 04:01:55.039120425 -0800
@@ -6669,16 +6669,34 @@
 "closerange(fd_low, fd_high)\n\n\
 Closes all file descriptors in [fd_low, fd_high), ignoring errors.");
 
+static int
+close_func(void *lohi, int fd)
+{
+    int lo = ((int *)lohi)[0];
+    int hi = ((int *)lohi)[1];
+
+    if (fd >= hi)
+        return (1);
+    else if (fd >= lo)
+        close(fd);
+
+    return (0);
+}
+
 static PyObject *
 posix_closerange(PyObject *self, PyObject *args)
 {
     int fd_from, fd_to, i;
+    int lohi[2];
+
     if (!PyArg_ParseTuple(args, "ii:closerange", &fd_from, &fd_to))
         return NULL;
     Py_BEGIN_ALLOW_THREADS
-    for (i = fd_from; i < fd_to; i++)
-        if (_PyVerify_fd(i))
-            close(i);
+
+    lohi[0] = fd_from;
+    lohi[1] = fd_to;
+    fdwalk(close_func, lohi);
+
     Py_END_ALLOW_THREADS
     Py_RETURN_NONE;
 }
@@ -8554,7 +8572,7 @@
     int name;
 
     if (PyArg_ParseTuple(args, "O&:sysconf", conv_sysconf_confname, &name)) {
-        int value;
+        long value;
 
         errno = 0;
         value = sysconf(name);
